java/maven/src/org/netbeans/modules/maven/debug/MavenJPDAStart.java

```diff
@@ -69,139 +72,126 @@
+        }
+        ClassPath jdkSourcePath = Utils.createJDKSourcePath(project);
 
-                    @Override
-                    public void run() {
-                        try {
-                            JPDADebugger.startListening(flc, args,
-                                                        new Object[]{properties, project});
-                        }
-                        catch (DebuggerStartException ex) {
-                            io.getErr().println("Debugger Start Error."); //NOI18N
-                            Logger.getLogger(JPDAStart.class.getName()).log(Level.INFO, "Debugger Start Error.", ex);
-                        }
+        if (getStopClassName() != null && getStopClassName().length() > 0) {
+            final MethodBreakpoint b = getStopMethod() != null ? Utils.createBreakpoint(getStopClassName(), getStopMethod()) : Utils.createBreakpoint(getStopClassName());
+            final Listener list = new Listener(b);
+            b.addPropertyChangeListener(MethodBreakpoint.PROP_VALIDITY, new PropertyChangeListener() {
+                @Override
+                public void propertyChange(PropertyChangeEvent pce) {
+                    if (Breakpoint.VALIDITY.INVALID.equals(b.getValidity()) && getStopMethod() != null) {
+                        //when the original method with method is not available (maybe defined in parent class?), replace it with a class breakpoint
+                        DebuggerManager.getDebuggerManager().removeBreakpoint(b);
+                        MethodBreakpoint b2 = Utils.createBreakpoint(getStopClassName());
+                        list.replaceBreakpoint(b2);
                     }
-                });
-            } catch (java.io.IOException ioex) {
-                io.getErr().println("IO Error:"); //NOI18N
-//                org.openide.ErrorManager.getDefault().notify(ioex);
-                lock[1] = ioex;
-            } catch (com.sun.jdi.connect.IllegalConnectorArgumentsException icaex) {
-                io.getErr().println("Illegal Connector"); //NOI18N
-                lock[1] = icaex;
-            } finally {
-                lock.notify();
-            }
+                }
+            });
+            DebuggerManager.getDebuggerManager().addDebuggerListener(
+                    DebuggerManager.PROP_DEBUGGER_ENGINES,
+                    list);
         }
+
+        final Map properties = new HashMap();
+        properties.put("sourcepath", sourcePath); //NOI18N
+        properties.put("name", getName()); //NOI18N
+        properties.put("jdksources", jdkSourcePath); //NOI18N
+        properties.put("baseDir", FileUtil.toFile(project.getProjectDirectory())); // NOI18N
+        if (RunUtils.isCompileOnSaveEnabled(project)) {
+            properties.put ("listeningCP", "sourcepath"); // NOI18N
+        }
+
+        final ListeningConnector flc = lc;
         
+        lastFuture = RP.submit(() -> {
```

Comment:
Flush the `lastFuture` to other threads, next `execute` will run in other than RP.